src/swc.rs

```diff
@@ -0,0 +1,282 @@
+use std::ops::Range;
+
+use swc_common::{BytePos, Span};
+use swc_ecma_parser::{Parser, StringInput};
+use swc_ecma_visit::swc_ecma_ast as ast;
+use swc_ecma_visit::{AstNodePath, VisitAstPath, VisitWithPath};
+
+use crate::scope_name::{NameComponent, ScopeName};
+use crate::Scopes;
+
+// TODO:
+// - getters / setters
+// - maybe even computed properties?
+
+pub fn parse_with_swc(src: &str) -> Scopes {
+    let syntax = tracing::trace_span!("parsing source").in_scope(|| {
+        let input = StringInput::new(src, BytePos(0), BytePos(src.len() as u32));
+
+        let mut parser = Parser::new(swc_ecma_parser::Syntax::default(), input, None);
+
+        parser.parse_module()
+    });
+    let syntax = match syntax {
+        Ok(syntax) => syntax,
+        Err(_) => return vec![],
```

Comment:
Should we surface the error somehow? Either in the return type or by logging it?